VogtObserver2/StockGrabber.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace VogtObserver2
+{
+    class StockGrabber : IObservable
+    {
+        private HistoricalDataInfo _hdi = new HistoricalDataInfo();
+        private List<HistoricalDataResponse> _hdr;
+
+        private List<IObserver> _observers = new List<IObserver>();
+
+        public void AddObserver(IObserver o)
+        {
+            _observers.Add(o);
+        }
+
+        public void RemoveObserver(IObserver o)
+        {
+            _observers.Remove(o);
+        }
+
+        public void FetchStockPrice(string symbol)
+        {
+            foreach (IObserver observer in _observers)
+            {
+                observer.Update(GetPrice(), symbol);
+            }
+        }
+
+        public double GetPrice()
+        {
+            
+            var temp = 0.0;
+
+            _hdr = _hdi.HistoricalData($"https://sandbox.iexapis.com/stable/stock/{Program.GetRandomStockSymbol()}/quote?token=Tpk_81485eef3d7041e6bd05ba956b85fa4e");
+
+            foreach (var historicaldata in _hdr)
```

Comment:
How are you sure that what you have is the latest price? Is it always ordered by date?

Might be worth doing a check on the **date**, if they have that in the response.

Consider the below. Do you see how this helps ensure you have the most recent stock? Now I havent read whether that API is **guarenteed or not** to return in order.

```c#
foreach(var historicaldata in _hdr) 
   if (historicaldata.DateChanged > tempDate) {
      tempValue = historicaldata.latestPrice;
   }
}
```